--- Giram-0.1.4/configure.in.wiget	Sun Dec 26 16:05:32 1999
+++ Giram-0.1.4/configure.in	Wed Jan 19 16:04:10 2000
@@ -310,14 +310,11 @@
                  ])
     ac_have_mesa_gl=$ac_cv_have_mesa_gl
     AC_MSG_RESULT($ac_have_mesa_gl)
-    if test "$ac_have_mesa_gl" = no ; then
-      gl_lib_1="GL"
-      GL_LIBS="-lGL -lGLU"
-    else
+    if test "$ac_have_mesa_gl" = yes; then
       AC_DEFINE(HAVE_MESA_GL)
-      gl_lib_1="MesaGL"
-      GL_LIBS="-lMesaGL -lMesaGLU"
     fi
+      gl_lib_1="GL"
+      GL_LIBS="-lGL -lGLU"
     # If it's MesaGL, we'd like to issue a warning if the version number
     # is less than or equal to 2.6, because that version had a security bug.
     #
