--- giram-0.3.5/configure.in.orig	2004-03-29 00:14:51.000000000 +0200
+++ giram-0.3.5/configure.in	2004-03-29 00:31:57.505575064 +0200
@@ -119,7 +119,7 @@
 AC_SUBST(GETTEXT_PACKAGE)
 dnl AC_DEFINE_UNQUOTED(GETTEXT_PACKAGE,"$GETTEXT_PACKAGE")
 
-ALL_LINGUAS="ca cs da de el en_GB es fi fr ga gl hu hr it ja ko nl no pl pt_BR ro ru sk sv tr uk zh_CN zh_TW"
+ALL_LINGUAS="ca cs da de el en_GB es fi fr ga gl hu hr it ja ko nl nb pl pt_BR ro ru sk sv tr uk zh_CN zh_TW"
 
 AC_PROG_INTLTOOL
 AM_GLIB_GNU_GETTEXT
